### PR TITLE
adding end_time attribute of RunInfo in search_runs

### LINKED ISSUE #6864: [FR] Add support for searching runs by end time
### Willingness to contribute

No. I cannot contribute this feature at this time.

### Proposal Summary

MLflow users can currently search for runs by the start time attribute, but they cannot search for runs by end time. This feature request proposes introducing the ability to search for runs based on end time by supplying a search filter string of the form `"attributes.end_time > 1663889102000`.

### Motivation

> #### What is the use case for this feature?

Users often want to filter runs by completion time, rather than start time, since some runs may not finish.

> #### Why is this use case valuable to support for MLflow users in general?

^

> #### Why is this use case valuable to support for your project(s) or organization?

^

> #### Why is it currently difficult to achieve this use case?

Not currently supported by MLflow's search APIs

### Details

This feature can be implemented by adding the `@searchable_attribute` decorator to https://github.com/mlflow/mlflow/blob/fa96efefc6f415d39038ce489aaf55217c6018af/mlflow/entities/run_info.py#L128-L130

Tests can be introduced to https://github.com/mlflow/mlflow/blob/master/tests/utils/test_search_utils.py.

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### LINKED ISSUE #6865: [DOC-FIX] Add start_time search examples to SearchRuns documentation
### Willingness to contribute

No. I cannot contribute a documentation fix at this time.

### URL(s) with the issue

https://mlflow.org/docs/latest/search-runs.html

### Description of proposal (what needs changing)

Add an example that demonstrates how to search for runs by start time to https://mlflow.org/docs/latest/search-runs.html. Right now, it's not obvious how this can be done.

Example filter string:

```
attributes.start_time > 1663889475000
```

Where `1663889475000` is milliseconds since the unix epoch.

### PR DESCRIPTION
**Pull request by subramaniam02**
Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

## Related Issues/PRs

#6864 
closes #6864
closes #6865 

## What changes are proposed in this pull request?

Support end_time in search_runs

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Support for end_time added to search_runs 

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by subramaniam02** [comment]
Coincidence my docs update will close this as well :)  #6865

File: docs/source/search-runs.rst [hunk 1]
```diff
@@ -86,7 +86,7 @@ that have a leading number. If an entity name contains a leading number, enclose
 Run Attributes
 ~~~~~~~~~~~~~~
 
-You can search using two run attributes contained in :py:class:`mlflow.entities.RunInfo`: ``status`` and ``artifact_uri``. Both attributes have string values. Other fields in ``mlflow.entities.RunInfo`` are not searchable.
+You can search using these run attributes contained in :py:class:`mlflow.entities.RunInfo`: ``status``, ``artifact_uri``, ``start_time``, ``end_time``. ``status``, ``artifact_uri`` attributes have string values. Other fields in ``mlflow.entities.RunInfo`` are not searchable.
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:89]:
```suggestion
You can search using the following run attributes contained in :py:class:`mlflow.entities.RunInfo`: ``status``, ``artifact_uri``, ``start_time``, ``end_time``. ``status`` and ``artifact_uri``. These attributes all have string values. Other fields in ``mlflow.entities.RunInfo`` are not searchable.
```

File: docs/source/search-runs.rst [hunk 2]
```diff
@@ -97,7 +97,10 @@ You can search using two run attributes contained in :py:class:`mlflow.entities.
 
 .. code-block:: sql
 
-  attributes.artifact_uri
+  attributes.artifact_uri = 'models:/mymodel/1'
+  attributes.status = 'ACTIVE'
+  attributes.start_time >= 1664067852747
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:102]: Can you add some text explaining to users that the search terms here are in unix timestamp format?

Comment (subramaniam02) [reply docs/source/search-runs.rst:103]: done

**Review by BenWilson2 — APPROVED** [review]
LGTM! Thanks for the addition :)

**Comment by anilravuri** [comment]
I still get this error:

mlflow.exceptions.RestException: INVALID_PARAMETER_VALUE: Could not parse search filter: `params.job_run_id != 'None' AND params.task='question-answering' AND attributes.start_time >= 1719772230423.086 AND attributes.end_time <= 1720031430423.086`. Invalid attribute column 'end_time' for filter_string. All valid attribute columns for filter_string are: 'status', 'start_time', 'artifact_uri', 'runId', 'run_name'

The thing here is - I'm trying to call a managed databricks instance. Does it make a difference?

I have the latest mlflow version: 2.14.2

**Pull request merged**